### ISSUE TITLE
Crash CocoaMQTTFramePublish

### ISSUE THREAD

**Issue by ariefnurputranto** [emqtt/CocoaMQTT#271]
Hi . . .
im using cocoaMQTT v1.2.1, sometime i getting crash in CocoaMQTTFramePublish

```
func send(_ frame: CocoaMQTTFramePublish) {
        guard let delegate = self.delegate else { return }
        delegate.dispatchQueue.async {
            delegate.deliver(self, wantToSend: frame)
        }
        
        // Insert to In-flight window for Qos1/Qos2 message
        if frame.qos != 0 && frame.msgid != nil {
            let _ = CocoaMQTTTimer.after(timeout) { [weak self] in
                guard let weakSelf = self else { return }
                printDebug("re-delvery frame \(frame)")
                weakSelf.send(frame)
            }
            inflight.append(frame)
        }
    }
```

crash inline "inflight.append(frame)"
![Screen Shot 2019-06-12 at 11 20 58](https://user-images.githubusercontent.com/12388348/59323259-2f08c400-8d04-11e9-8ddf-aa39241835ce.png)

![Screen Shot 2019-06-12 at 14 06 10](https://user-images.githubusercontent.com/12388348/59330360-73ec2500-8d1b-11e9-9d07-7eeb157ddf61.png)

**Comment by HJianBo** [comment]
Hi, @ariefnurputranto Thanks for your feedback!

I have fixed it and it will be released at v1.2.2 for a hot fix version!

**Comment by ariefnurputranto** [comment]
thanks @HJianBo, btw when we get the v1.2.2? because in your milestones 1.2.2 not add release date

**Comment by HJianBo** [comment]
Hi, @ariefnurputranto  we have released 1.2.2 to fix this bug!

Please check it if you can. Thanks!

**Comment by ariefnurputranto** [comment]
Thanks, will check soon @HJianBo
